shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1428,7 +1434,9 @@ void WebContents::RenderFrameDeleted(
   //
 
   // clear out objects that have been granted permissions
-  granted_devices_.clear();
+  if (!granted_devices_.empty()) {
+    granted_devices_[render_frame_host->GetFrameTreeNodeId()].clear();
```

Comment:
This still lets `granted_devices_` grow without bound (it will be full of empty values). Instead,

```suggestion
    granted_devices_.erase(render_frame_host->GetFrameTreeNodeId());
```